jbpm-task-assigning/jbpm-task-assigning-core/src/test/java/org/jbpm/task/assigning/model/solver/realtime/AssignTaskProblemFactChangeTest.java

```diff
@@ -0,0 +1,320 @@
+        programmedChanges.add(new ProgrammedAssignTaskProblemFactChange(task, user));
+
+        //repeat assign Task_15 to User_5
+        task = solution.getTaskList().get(15);
+        user = solution.getUserList().get(5);
+        programmedChanges.add(new ProgrammedAssignTaskProblemFactChange(task, user));
+
+        long nextTaskId = solution.getTaskList().stream()
+                .mapToLong(Task::getId)
+                .max().orElse(-1) + 1;
+
+        //assign a brand new task "NewTask_x and assign to User_0
+        user = solution.getUserList().get(0);
+        task = new Task(nextTaskId, "NewTask_" + nextTaskId, 1);
+        programmedChanges.add(new ProgrammedAssignTaskProblemFactChange(task, user));
+
+        //assign a brand new task "NewTask_x and assign to User_2
+        nextTaskId++;
+        user = solution.getUserList().get(2);
+        task = new Task(nextTaskId, "NewTask_" + nextTaskId, 1);
+        programmedChanges.add(new ProgrammedAssignTaskProblemFactChange(task, user));
+
+        //assign a brand new task "NewTask_x and assign to User_5
+        nextTaskId++;
+        user = solution.getUserList().get(5);
+        task = new Task(nextTaskId, "NewTask_" + nextTaskId, 1);
+        programmedChanges.add(new ProgrammedAssignTaskProblemFactChange(task, user));
+
+        assignTaskProblemFactChangeTest(solution, solutionResource, FIXED_TEST, programmedChanges);
+    }
+
+    private void assignTaskProblemFactChangeRandomChangeSetTest(String solutionResource) throws Exception {
+        TaskAssigningSolution solution = readTaskAssigningSolution(solutionResource);
+        solution.getUserList().add(User.PLANNING_USER);
+
+        int taskCount = solution.getTaskList().size();
+        int userCount = solution.getUserList().size();
+        int randomChanges = taskCount / 2 + random.nextInt(taskCount / 2);
+
+        //prepare the list of changes to program
+        List<ProgrammedAssignTaskProblemFactChange> programmedChanges = new ArrayList<>();
+
+        Task randomTask;
+        User randomUser;
+        for (int i = 0; i < randomChanges; i++) {
+            randomTask = solution.getTaskList().get(random.nextInt(taskCount));
+            randomUser = solution.getUserList().get(random.nextInt(userCount));
+            programmedChanges.add(new ProgrammedAssignTaskProblemFactChange(randomTask, randomUser));
+        }
+        assignTaskProblemFactChangeTest(solution, solutionResource, RANDOM_TEST, programmedChanges);
+    }
+
+    private void assignTaskProblemFactChangeTest(TaskAssigningSolution solution,
+                                                 String solutionResource,
+                                                 String testType,
+                                                 List<ProgrammedAssignTaskProblemFactChange> programmedChanges) throws Exception {
+        TaskAssigningSolution initialSolution = executeSequentialChanges(solution, programmedChanges);
+        if (writeTestFiles()) {
```

Comment:
soft suggestion: let's move writing the files into a separate method in order not to pollute this method with unnecessary details.